pkg/diagnostics/client/node_definitions.go

```diff
@@ -0,0 +1,100 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types/diagnostic"
+)
+
+const (
+	clientErrorGettingNodes = `Client error while retrieving node records. Client retrieved records
+during discovery, so this is likely to be a transient error. Try running
+diagnostics again. If this message persists, there may be a permissions
+problem with getting node records. The error was:
+
+(%T) %[1]v`
+
+	nodeNotReady = `Node {{.node}} is defined but is not marked as ready.
+Ready status is {{.status}} because "{{.reason}}"
+If the node is not intentionally disabled, check that the master can
+reach the node hostname for a health check and the node is checking in
+to the master with the same hostname.
+
+While in this state, pods should not be scheduled to deploy on the node,
+and any existing scheduled pods will be considered failed and removed.
+`
+)
+
+// NodeDefinitions
+type NodeDefinition struct {
+	KubeClient *kclient.Client
+
+	Log *log.Logger
+}
+
+func (d NodeDefinition) Description() string {
+	return "Check node records on master"
+}
+func (d NodeDefinition) CanRun() (bool, error) {
+	if d.KubeClient == nil {
+		// TODO make prettier?
+		return false, errors.New("must have kube client")
+	}
+	if _, err := d.KubeClient.Nodes().List(labels.LabelSelector{}, fields.Everything()); err != nil {
```

Comment:
> But this isn't reusable, and it does the same thing as you're going to have to do again in a moment at the diagnostic.

It's not intended to be.  Different `Diagnostics` require different levels of permissions and only the diagnostic author reasonably knows what they are.  It's rather unlikely that two diagnostics will be looking for the same levels of permissions.  I saw value to the skip you were going before, but it should be based on what permissions are required for the test.  I could even see switching this to a subject access review.